refactor the examples

wopeizl opened the pull request:
1. move the original examples into fit_a_line sub folder
2. add the ctr exampel
test=develop

wopeizl committed:
refactor the examples

typhoonzero commented on example/ctr/script/k8s_tools.py:
```diff
@@ -0,0 +1,181 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
If changes to this file and `paddle_k8s` can be compatible with all jobs, can just change these files in the original place, try not to keep multiple copies of this file

wopeizl replied:
ok

typhoonzero commented on example/ctr/README:
```diff
@@ -0,0 +1,18 @@
+
+CTR分布式训练
```
better to write this using "markdown" format, and write the steps for use

wopeizl replied:
ok

wopeizl committed:
add the rst instruction
remove the files under script and merge the functions into edl/docker
test=develop

typhoonzero commented on docker/paddle_k8s:
```diff
@@ -62,27 +62,33 @@ check_trainer_ret() {
-  export PADDLE_PSERVER_IPS=$(python /root/k8s_tools.py fetch_ips ${pserver_label} ${PADDLE_INIT_PORT})
-  export PADDLE_TRAINER_IPS=$(python /root/k8s_tools.py fetch_ips ${trainer_label})
+  export PADDLE_PSERVERS=$(python /root/k8s_tools.py fetch_endpoints ${pserver_label} ${PADDLE_PORT})
+  # export PADDLE_TRAINER_IPS=$(python /root/k8s_tools.py fetch_ips ${trainer_label})
```
why comment out this line?

wopeizl replied:
restore

typhoonzero commented on docker/paddle_k8s:
```diff
@@ -62,27 +62,33 @@ check_trainer_ret() {
   if [ "${PADDLE_TRAINING_ROLE}" == "TRAINER" ] || [ "${PADDLE_TRAINING_ROLE}" == "WORKER" ]; then
     check_failed_cnt 1
     task_index=$(python /root/k8s_tools.py fetch_id ${trainer_label})
+    # download training materials for trainer task_index=0-9
```
download procedure can be put into train.py

wopeizl replied:
done

typhoonzero commented on example/ctr/ps-train/trainer.yaml:
```diff
@@ -0,0 +1,56 @@
+        env:
+        - {name: GLOG_v, value: '0'}
+        - {name: GLOG_logtostderr, value: '1'}
+        #- {name: GLOG_vmodule, value: 'executor=4,operator=4'}
```
remove commented out line or add description

wopeizl replied:
ok

wopeizl committed:
move the download process to train script test=develop

typhoonzero commented:
Force merged, I'll fix the build issue on develop branch later.
